bpf/mock/check_headers.sh

```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2021 Authors of Cilium
+
+# script to check if helpers.h, helpers_skb.h, and helpers_xdp.h are up to date
+
+#!/bin/bash
+
+sed '/#ifndef BPF_/,/#endif/d; /#if __ctx_is/,/#endif/d; s/"compiler\.h"/<bpf\/compiler\.h>/g; /ctx\/ctx\.h/d; s/__BPF_HELPERS\(.*__\)/__MOCK_HELPERS\1/g; s/static //g; s/BPF_\w*(\(\w*\), /\1(/g; s/BPF_\w*(\(\w*\)/\1(/g; s/__printf(.*) //g; / =/{ N; s/ =.*;/;/ };' ../include/bpf/helpers.h > helpers_temp.h
+sed '/#ifndef BPF_/,/#endif/d; /#if __ctx_is/,/#endif/d; s/"features_skb\.h"/<bpf\/features_skb\.h>/g; s/"compiler\.h"/<bpf\/compiler\.h>/g; /"helpers\.h"/d; s/__BPF_HELPERS\(.*__\)/__MOCK_HELPERS\1/g; s/static //g; s/BPF_\w*(\(\w*\), /\1(/g; s/BPF_\w*(\(\w*\)/\1(/g; s/__printf(.*) //g; / =/{ N; s/ =.*;/;/ };' ../include/bpf/helpers_skb.h > helpers_skb_temp.h
+sed '/#ifndef BPF_/,/#endif/d; /#if __ctx_is/,/#endif/d; s/"features_xdp\.h"/<bpf\/features_xdp\.h>/g; s/"compiler\.h"/<bpf\/compiler\.h>/g; /"helpers\.h"/d; s/__BPF_HELPERS\(.*__\)/__MOCK_HELPERS\1/g; s/static //g; s/BPF_\w*(\(\w*\), /\1(/g; s/BPF_\w*(\(\w*\)/\1(/g; s/__printf(.*) //g; / =/{ N; s/ =.*;/;/ };' ../include/bpf/helpers_xdp.h > helpers_xdp_temp.h
+
+if cmp helpers_temp.h helpers.h
+then echo mock/helpers.h is up to date!
+else echo mock/helpers.h is out of date! Please run generate_headers.sh or manually add the new helpers to mock/helpers.h and run mock_helpers.sh to generate mock libraries.
```

Comment:
maybe we can add a Makefile target to call these scripts and then devs can simple run "make XXX"?

Reply:
right, I think that the `sed` portions of this can potentially stay in a `.sed` file but the comparison / make pieces seem like they could be folded into a make target.